src/database/Matrimonial/migrations/V003_20241006_Add CreateUser and delete and update.sql

```diff
@@ -1,104 +1,162 @@
+    SET @p_ErrorTable = JSON_ARRAY_APPEND(@p_ErrorTable, '$', 'Address can not be null or empty');
     END IF;
-
-    -- Insert valid data into Users table
+    
+    SELECT @p_ErrorTable;
+    
+    IF JSON_LENGTH(@p_ErrorTable) > 0 THEN
+    SET p_Errors = JSON_UNQUOTE(@p_ErrorTable);
+    SIGNAL SQLSTATE '45000'
+    SET MESSAGE_TEXT = p_Errors;
+    END IF;
+    
     INSERT INTO Users (FirstName, LastName, DOB, Gender, Email, Phone, Address)
-    VALUES (p_FirstName, p_LastName, p_DOB, p_Gender, p_Email, p_Phone, p_Address);
+    VALUES(p_FirstName, p_LastName, p_DOB, p_Gender, p_Email, p_Phone, p_Address);
+    
+    SELECT last_insert_id() INTO p_Errors;
+    
 
-    -- Return new user ID
-    SET p_UserId = LAST_INSERT_ID();
 END //
 
-CREATE PROCEDURE UpdateUser(
+CREATE  PROCEDURE Update_Users(
+	
     IN p_UserId INT,
     IN p_FirstName VARCHAR(255),
-    IN p_LastName VARCHAR(255),
+    IN p_LastName varchar(255),
     IN p_DOB DATE,
     IN p_Gender VARCHAR(255),
-    IN p_Email NVARCHAR(255),
-    IN p_Phone BIGINT,
-    IN p_Address NVARCHAR(255)
+    IN p_Email VARCHAR(255),
+    IN p_Phone VARCHAR(255),
+    IN p_Address VARCHAR(255),
+    OUT p_Errors VARCHAR(255)
+    
 )
 BEGIN
-    -- Ensure user exists
-    DECLARE user_exists INT;
-    SELECT COUNT(*) INTO user_exists FROM Users WHERE UserId = p_UserId;
-
-    IF user_exists = 0 THEN
-        SIGNAL SQLSTATE '45000' SET MESSAGE_TEXT = 'User not found';
-    ELSE
-        -- Update user information
-        UPDATE Users
-        SET
-            FirstName = p_FirstName,
-            LastName = p_LastName,
-            DOB = p_DOB,
-            Gender = p_Gender,
-            Email = p_Email,
-            Phone = p_Phone,
-            Address = p_Address
-        WHERE UserId = p_UserId;
 
-        -- Check if any rows were affected
-        IF ROW_COUNT() = 0 THEN
-            SIGNAL SQLSTATE '45000' SET MESSAGE_TEXT = 'No changes made';
-        END IF;
+	-- initiaize error table
+    SET @p_ErrorTable = '[]';
+    
+    -- add validation
+    IF p_FirstName IS NULL OR p_FirstName = '' THEN
+    SET @p_ErrorTable = JSON_ARRAY_APPEND(@p_ErrorTable, '$', 'First name cant be null or empty');
+    END IF;
+    
+    IF p_LastName IS NULL OR p_LastName = '' THEN 
+    SET @p_ErrorTable = JSON_ARRAY_APPEND(@p_ErrorTable, '$', 'last name cant be null or empty');
+    END IF;
+    
+    IF p_DOB IS NULL THEN 
+    SET @p_ErrorTable = JSON_ARRAY_APPEND(@p_ErrorTable, '$', 'DOB cant be null or empty');
+    END IF;
+    
+    IF  p_Gender!= 'Male' AND  p_Gender != 'Female' THEN
+    SET @p_ErrorTable = JSON_ARRAY_APPEND(@p_ErrorTable, '$', 'Gender cant be null or empty');
+    END IF;
+    
+    IF p_Email IS NULL OR p_Email = '' OR NOT p_Email REGEXP '^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,4}$' THEN
+    SET @p_ErrorTable = JSON_ARRAY_APPEND(@p_ErrorTable, '$', 'Invalid Email ');
+    END IF;
+    
+    IF p_Phone IS NULL OR p_Phone = '' THEN
```

Comment:
we can use regex here as well

https://stackoverflow.com/questions/2113908/what-regular-expression-will-match-valid-international-phone-numbers

\+(9[976]\d|8[987530]\d|6[987]\d|5[90]\d|42\d|3[875]\d|
2[98654321]\d|9[8543210]|8[6421]|6[6543210]|5[87654321]|
4[987654310]|3[9643210]|2[70]|7|1)\d{1,14}$